By default gperf puts a header into generated files with the full path to
the tool along with the commandline used. This patch removes the path to
the binary, allowing reproducible source files (which can be included in
debug source packages).

Upstream-Status: Pending
Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>

Index: gperf-3.1/src/options.cc
===================================================================
--- gperf-3.1.orig/src/options.cc
+++ gperf-3.1/src/options.cc
@@ -280,6 +280,13 @@ Options::print_options () const
     {
       const char *arg = _argument_vector[i];
 
+      if (i == 0) {
+          const char *shortarg = strrchr(arg, '/');
+          if (shortarg) {
+              arg = shortarg + 1;
+          }
+      }
+
       /* Escape arg if it contains shell metacharacters.  */
       if (*arg == '-')
         {
